designer/server/src/main/scala/pl/touk/nussknacker/ui/definition/DefinitionsService.scala

```diff
@@ -97,6 +102,39 @@ class DefinitionsService(
     }
   }
 
+  // Components are fetched from cache, as long as the value in cache exists and fragment definitions did not change since last cache update.
+  // Otherwise, the cache in updated.
+  private def getComponents(
+      processingType: ProcessingType,
+      forFragment: Boolean,
+      fragments: List[CanonicalProcess],
+  ): Components = {
+    Option(fragmentComponentsCache.getIfPresent((processingType, forFragment))) match {
+      case Some((cachedFragments, cachedComponents)) if cachedFragments == fragments =>
```

Comment:
Are you sure that it is fast? You compare large objects (scenarios). In case if they are equal, every single node and their parameters will be compared.

Reply:
No longer applicable, cache removed, description of this PR updated.